server/src/main/java/org/folio/shared/index/api/SharedIndexService.java

```diff
@@ -85,44 +83,50 @@ static String getQueryParameter(RequestParameters params) {
     PgCqlQuery pgCqlQuery = getPqCqlQueryForRecords();
     RequestParameters params = ctx.get(ValidationHandler.REQUEST_CONTEXT_KEY);
     pgCqlQuery.parse(getQueryParameter(params));
-    String m = getParameterString(params.queryParameter("matchkeyid"));
     Storage storage = new Storage(ctx);
-    if (m != null) {
-      int maxIterations = getParameterInteger(
-          params.queryParameter("maxiterations"), MATCH_MAX_ITERATIONS);
-      List<String> matchKeyIds = Arrays.asList(m.split(","));
-      return storage.getCluster(pgCqlQuery.getWhereClause(), matchKeyIds, maxIterations)
-          .map(records -> {
-            JsonArray items = new JsonArray();
-            records.forEach((k, v) -> items.add(v));
-            JsonObject result = new JsonObject();
-            result.put("items", items);
-            result.put("totalRecords", records.size());
-            HttpResponse.responseJson(ctx, 200).end(result.encode());
-            return null;
-          });
-    }
     return storage.getSharedRecords(ctx, pgCqlQuery.getWhereClause(),
         pgCqlQuery.getOrderByClause());
   }
 
+  Future<Void> getSharedRecordGlobalId(RoutingContext ctx) {
+    RequestParameters params = ctx.get(ValidationHandler.REQUEST_CONTEXT_KEY);
+    String id = getParameterString(params.pathParameter("globalId"));
+    Storage storage = new Storage(ctx);
+    return storage.selectSharedRecord(id)
+        .onSuccess(res -> {
+          if (res == null) {
+            HttpResponse.responseError(ctx, 404, id);
+            return;
+          }
+          HttpResponse.responseJson(ctx, 200).end(res.encode());
+        })
+        .mapEmpty();
+  }
+
   Future<Void> getClusters(RoutingContext ctx) {
     PgCqlQuery pgCqlQuery = getPqCqlQueryForRecords();
+    pgCqlQuery.addField(
+        new PgCqlField("cql.allRecords", PgCqlField.Type.ALWAYS_MATCHES));
+    pgCqlQuery.addField(
+        new PgCqlField("match_value", "matchValue", PgCqlField.Type.TEXT));
```

Comment:
Am I correct that with a request like ?matchkeyid=id1,id2&query=matchValue=Smith we will get clusters back where matchValue matches for id1 OR id2?

Reply:
Looking on the code I see that the param accepts only one matchkeyid value, not a CSV

Reply:
Correct (not supported). I guess you'd like the CSV support.

Reply:
Actually, I think this is sufficient.